cmd/lotus-shed/cid.go

```diff
@@ -83,3 +91,41 @@ var cidIdCmd = &cli.Command{
 		return nil
 	},
 }
+
+var cidFromCarCmd = &cli.Command{
+	Name:      "manifest-cid-from-car",
+	Usage:     "Get the manifest CID from a car file",
+	ArgsUsage: "[path]",
+	Action: func(cctx *cli.Context) error {
+		ctx := context.TODO()
+
+		cf := cctx.Args().Get(0)
+		f, err := os.OpenFile(cf, os.O_RDONLY, 0664)
+		if err != nil {
+			return xerrors.Errorf("opening the car file: %w", err)
+		}
+
+		bs := blockstore.NewMemory()
+		if err != nil {
+			return err
+		}
+
+		data, err := io.ReadAll(f)
+		if err != nil {
+			return xerrors.Errorf("error reading car file: %w", err)
+		}
+
+		blobr := bytes.NewReader(data)
+
+		hdr, err := car.LoadCar(ctx, bs, blobr)
```

Comment:
No need to do a `io.ReadAll` here and then create a new `io.Reader` as `os.FIle` already implements `io.Reader` which can be passed directly into `car. LoadCar`.